### PR TITLE
Move install docs to the website, add "moved" notices

### PR DESCRIPTION
**Pull request by ahmetb**
* refactor README, USER_GUIDE, DEVELOPER_GUIDE with notes indicating they are
  moved to the website.

* move installation instructions from README to the user guide.

Related issue: #515

### CODE REVIEW

**Comment by corneliusweig** [comment]
We need to coordinate this with https://github.com/kubernetes/kubernetes/pull/88593

File: docs/NAMING_GUIDE.md [hunk 1]
```diff
@@ -1,78 +1,4 @@
...
-readability and discoverability of a plugin more than it is saving keystrokes.
-
-- **DON'T:** `kubectl new-ing`: Hard to spot and the plugin is for
-  Ingress.<br/>
-  **DO:** `kubectl debug-ingress`.
-
------
-
-If you have suggestions to this guide, open an issue or send a pull request.
+This page is moved to our website:
```
Comment (corneliusweig) [inline docs/NAMING_GUIDE.md]:
```suggestion
This page has moved to our website:
```

File: docs/USER_GUIDE.md [hunk 2]
```diff
@@ -1,134 +1,4 @@
...
-
-
-## Disabling update checks
-
-When using krew, it will occasionally check if a new version of krew is available
-by calling the GitHub API. If you want to opt out of this feature, you can set
-the `KREW_NO_UPGRADE_CHECK` environment variable. To permanently disable this,
-add the following to your `~/.bashrc`, `~/.bash_profile`, or `~/.zshrc`:
-
-    export KREW_NO_UPGRADE_CHECK=1
```
Comment (corneliusweig) [inline docs/USER_GUIDE.md]: This is completely gone now. I think we should document this somewhere on our user guide.

File: docs/USER_GUIDE.md [hunk 3]
```diff
@@ -1,134 +1,4 @@
...
-
-## Uninstalling Krew
-
-Uninstalling `krew` is as easy as deleting its installation directory.
-
-To find `krew`'s installation directory, run:
-
-    kubectl krew version
-
-And delete the directory listed in `BasePath:` field. On macOS/Linux systems,
```
Comment (corneliusweig) [inline docs/USER_GUIDE.md]: Our docs site doesn't mention how to discover the correct basePath via `krew version`. Should we add it back?

File: docs/DEVELOPER_GUIDE.md [hunk 4]
```diff
@@ -1,349 +1,4 @@
...
-
-    chmod +x ./kubectl-foo
-
-(Since this plugin requires `bash`, it will only work on Unix platforms. If you
-need to support windows, develop a `kubectl-foo.exe` executable.)
-
-Now, if you place this plugin to anywhere in your `$PATH` directories, you
-should be able to call it like:
-
-    kubectl foo
```
Comment (corneliusweig) [inline docs/DEVELOPER_GUIDE.md]: This section is gone from our new docs. I think it's worth adding a new page with something like this.

File: docs/DEVELOPER_GUIDE.md [hunk 5]
```diff
@@ -1,349 +1,4 @@
...
-
-### Packaging plugins for krew
-
-To package a plugin via krew, you need to:
-
-1. Develop a plugin and archive it into a `.zip` or `.tar.gz`.
-2. Write a [plugin manifest file](#writing-a-plugin-manifest).
-3. Test installation locally with the archive and the manifest file.
```
Comment (corneliusweig) [inline docs/DEVELOPER_GUIDE.md]: Our new docs are far less explicit about packaging a plugin. Do you think it's worth being more verbose like here?

File: docs/DEVELOPER_GUIDE.md [hunk 6]
```diff
@@ -1,349 +1,4 @@
...
-### Writing a plugin manifest
-
-Each krew plugin has a "plugin manifest" file that lives in the [krew index
-repository][index].
-
-This file describes which files krew must copy out of your plugin archive and
-how to run your plugin. It is also used to determine if an upgrade to your
-plugin is available and how to install it.
-
-For a plugin named `foo`, the plugin manifest file should be named `foo.yaml`.
```
Comment (corneliusweig) [inline docs/DEVELOPER_GUIDE.md]: > For a plugin named `foo`, the plugin manifest file should be named `foo.yaml`.

This is far less explicit in our new docs. Is it worth being more verbose here?

File: docs/DEVELOPER_GUIDE.md (hunk 5 above)
Comment (ahmetb) [reply docs/DEVELOPER_GUIDE.md]: Yeah I agree. We need a small "distributing with Krew" page.

File: docs/DEVELOPER_GUIDE.md (hunk 4 above)
Comment (ahmetb) [reply docs/DEVELOPER_GUIDE.md]: This is at http://localhost:1313/docs/developer-guide/develop/plugin-development/

File: docs/DEVELOPER_GUIDE.md (hunk 5 above)
Comment (ahmetb) [reply docs/DEVELOPER_GUIDE.md]: Explained in f7daf30.

File: docs/USER_GUIDE.md (hunk 3 above)
Comment (ahmetb) [reply docs/USER_GUIDE.md]: I think we're fine.
Anyone using custom path is probably a developer who knows how to find the right path.

File: docs/USER_GUIDE.md (hunk 2 above)
Comment (ahmetb) [reply docs/USER_GUIDE.md]: Good catch, added in 6b86b8a as a new page.

File: docs/DEVELOPER_GUIDE.md (hunk 6 above)
Comment (ahmetb) [reply docs/DEVELOPER_GUIDE.md]: Addressed in 756b221

File: docs/DEVELOPER_GUIDE.md (hunk 4 above)
Comment (corneliusweig) [reply docs/DEVELOPER_GUIDE.md]: Ah right. O_O

**Pull request merged**